File: examples/mq/publish/go/publishAMessage.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/mq"
)

// Publish a message. Specify a topic to group messages for a specific topic.
func PublishAmessage() {
	mqService := mq.NewMqService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := mqService.Publish(&mq.PublishRequest{
		Message: map[string]interface{}{
			"type": "signup",
			"user": "john",
			"id":   "1",
		},
		Topic: "events",
	})
	fmt.Println(rsp, err)
}

File: examples/mq/publish/node/publishAMessage.js
const { MqService } = require("m3o/mq");

// Publish a message. Specify a topic to group messages for a specific topic.
async function publishAmessage() {
  let mqService = new MqService(process.env.MICRO_API_TOKEN);
  let rsp = await mqService.publish({
    message: {
      id: "1",
      type: "signup",
      user: "john",
    },
    topic: "events",
  });
  console.log(rsp);
}

publishAmessage();
